### PR TITLE
feature(forms): rewrites friendspicker and collections

### PR DESCRIPTION
**Pull request by hypeJunction**
Rewrties friendspicker into a usable form input disconnected from friends collections
management interface. Refactors friends collections management interface and removes
all the boilerplate code.
- [ ] Isolate friendspicker CSS into a separate view shared by core, aalboard and admin
- [x] Update Notifications settings table to not use old friendspicker markup/js (#10287)

### CODE REVIEW

**Comment by hypeJunction** [comment]
I imagine this will have to wait until 3.x, but throwing it out there. this rewrites the much-loved friendspicker into a usable form element.

collections
friendspicker

File: views/default/core/friends/collection.php [hunk 1]
```diff
@@ -1,67 +1,55 @@
...
-	$count = sizeof($vars['collection']->members);
-} else {
-	$count = 0;
+$collection = elgg_extract('collection', $vars);
+if (empty($collection)) {
+	return;
 }
 
-echo "<li><h2>";
+$members = get_members_of_access_collection($collection->id) ? : array();
```
Comment (beck24) [inline views/default/core/friends/collection.php]: If we're re-writing this we should get away from outputting *all* friends.  These are trouble spots for scalability that we're constantly having to override in our projects become someone thinks it's cool to friend 10k people.

Comment (mrclay) [inline views/default/core/friends/collection.php]: What do you do? Paginate?

Comment (beck24) [inline views/default/core/friends/collection.php]: Yeah, ajax pagination or infinite scroll

Comment (hypeJunction) [inline views/default/core/friends/collection.php]: Agreed, but then it wouldn't be a friends picker per se.
It's good UX for anything under 100 items. With collections, we should
perhaps drop the 'edit' form and use 'add members' form, and implement a
'remove member' option.

**Comment by juho-jaakkola** [comment]
Looks good, but so far I haven't had time to review.

**Comment by mrclay** [comment]
I rebased this. If you want it:
```
git remote add mrclay git@github.com:mrclay/Elgg-leaf.git
git fetch mrclay
git reset --hard mrclay/hypeJunction-friendspicker
```

**Comment by hypeJunction** [comment]
I am going to make it a plugin I think, and we can then polish it up if we want it in core.

**Comment by jdalsem** [comment]
I think this should definitively end up in core... multiple plugins use the friendspicker so this would benefit a lot of people..

**Comment by hypeJunction** [comment]
I am reviving this for 3.0. If there are any objects, please voice them.

**Comment by jdalsem** [comment]
Also fixes https://github.com/Elgg/Elgg/issues/7463

**Comment by hypeJunction** [comment]
@Elgg/contributors cast your vote: https://elgg.org/discussion/view/2551792/kill-the-friendspicker

**Comment by hypeJunction** [comment]
Replaced by #10307

**Pull request closed without merging**